### ISSUE TITLE
Rule proposal: prefer-optional-chaining

### ISSUE THREAD

**Issue by axetroy** [sindresorhus/eslint-plugin-unicorn#2434]
### Description

for clean code

### Fail

```js
callback && callcak()
```

### Pass

```js
callback?.()
```

### Proposed rule name

prefer-optional-chaining

### Additional Info

Optional Chaining should have other scenarios, I will add them later

**Comment by sindresorhus** [comment]
Duplicate of #429
